src/proto/streams/counts.rs

```diff
@@ -60,11 +60,13 @@ impl Counts {
     /// # Panics
     ///
     /// Panics on failure as this should have been validated before hand.
-    pub fn inc_num_recv_streams(&mut self) {
+    pub fn inc_num_recv_streams(&mut self, stream: &mut store::Ptr) {
         assert!(self.can_inc_num_recv_streams());
+        assert!(!stream.is_counted);
```

Comment:
This question applies for most of the new asserts: do we want to be adding new `assert`s everywhere, or just `debug_assert`? Is it better for this to crash in production?

Reply:
I think it depends on the severity of the assertion failure.

If this fails, then we could get into situations where subtractions wrap around. I think this should be guarded again?

Reply:
Any further thoughts on this?

Reply:
Hm, so of course, crashing in production is a major bummer, as it destroys **all** in progress requests... In this specific case, it seems like if this assert were to fail, then the subtraction would eventually underflow and crash anyways. So perhaps this is fine...

(It'd probably be neat if we could just encode this in the types somehow instead, but :shrug:)

Reply:
Yeah.... ideally it would only invalidate the connection.

That said, we have a whole bunch of existing `assert!` statements. Maybe we should punt this as an open question: "how to better deal w/ internal errors".